packetizer.go

```diff
@@ -1,64 +1,89 @@
 package rpc
 
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	"github.com/ugorji/go/codec"
+)
+
 type packetizer interface {
-	Packetize() error
+	NextFrame() (RPCMessage, error)
 }
 
 type packetHandler struct {
-	receiver receiver
-	dec      byteReadingDecoder
+	dec           decoder
+	reader        io.Reader
+	frameDecoders decoderPool
+	protocols     *protocolHandler
+	calls         *callContainer
+	mtx           sync.Mutex
```

Comment:
Oh yes, I added it to ensure no other goroutines could interrupt the stream between reading the length of the packet and the packet itself. That said I think you're right, it's done synchronously and thus doesn't need a mutex.